node/src/components/rpc_server/rpcs/chain.rs

```diff
@@ -468,6 +475,111 @@ impl RpcWithOptionalParamsExt for GetEraInfoBySwitchBlock {
     }
 }
 
+/// Params for "chain_get_era_summary" RPC response.
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct GetEraSummaryParams {
+    /// The block identifier.
+    pub block_identifier: BlockIdentifier,
+}
+
+impl DocExample for GetEraSummaryParams {
+    fn doc_example() -> &'static Self {
+        &*GET_ERA_SUMMARY_PARAMS
+    }
+}
+
+/// Result for "chain_get_era_summary" RPC response.
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+#[serde(deny_unknown_fields)]
+pub struct GetEraSummaryResult {
+    /// The RPC API version.
+    #[schemars(with = "String")]
+    pub api_version: ProtocolVersion,
+    /// The era summary.
+    pub era_summary: EraSummary,
+}
+
+impl DocExample for GetEraSummaryResult {
+    fn doc_example() -> &'static Self {
+        &*GET_ERA_SUMMARY_RESULT
+    }
+}
+
+/// "chain_get_era_summary" RPC
+pub struct GetEraSummary {}
+
+impl RpcWithOptionalParams for GetEraSummary {
+    const METHOD: &'static str = "chain_get_era_summary";
```

Comment:
Please remember to extend out RPC testing suite in `casper-test`. Could (and should) be done in a separate PR, but please request a proper ticket.

Reply:
@RitaMAllenCA pls add ticket for this